flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -150,16 +126,13 @@
 	 */
 	private boolean nfaChanged;
 
-	public NFA(
-			final TypeSerializer<T> eventSerializer,
+	public NFA(final TypeSerializer<T> eventSerializer,
 			final long windowTime,
 			final boolean handleTimeout) {
-
 		this.eventSerializer = eventSerializer;
```

Comment:
this PR does not change anything in this manner (previously it also should not be null), but I agree we can add this check

Reply:
I checked the `NFA` code and currently in master it seems like it could handle `null` on this parameter while after this change that's no longer the case? But regardless as you said `checkNotNull` is a good thing anyway.

Reply:
The difference in handling `null` seems strange to me. Could you tell me how did you check it?

Anyway as I already said will add the check.

Reply:
If you check all of the `NFA`'s `eventSerializer` usages, there are all in the `extractCurrentMatches` method, which before your change was guarding against `(eventSerializer == null)` condition. Thus (maybe unintentionally) this class's `evenSerializer` field could be `@Nullable` or was supporting `new NFA(null, 1, false)` calls. From what I have checked (by following up all of the constructor invocations), there seems to be no code path for such construction to happen at the moment, but adding `checkNotNull(eventSerializer)` will guard against such mistakes in the future.